ros/angel_utils/scripts/bag_extractor.py

```diff
@@ -0,0 +1,385 @@
+            self.num_total_msgs += 1
+
+            if (self.num_total_msgs % 100) == 0:
+                self.log.info(f"Parsing message: {self.num_total_msgs}")
+
+            # Attempt to call the message handler for this message type
+            try:
+                self.msg_type_to_handler_map[type(msg)](msg)
+            except KeyError as e:
+                pass
+
+        # Print bag stats
+        self.print_bag_info()
+
+        # Save the audio file
+        self.create_wav_file()
+
+        # Write the json data to disk
+        if self.extract_eye_gaze_data:
+            eye_gaze_file = self.data_folder + "eye_gaze_data.txt"
+            with open(eye_gaze_file, mode="w", encoding='utf-8') as f:
+                json.dump(self.eye_gaze_msgs, f)
+            self.log.info(f"Created eye gaze file: {eye_gaze_file}")
+        else:
+            self.log.info(f"Skipping eye gaze file creation")
+
+        if self.extract_hand_pose_data:
+            hand_pose_file = self.data_folder + "hand_pose_data.txt"
+            with open(hand_pose_file, mode="w", encoding='utf-8') as f:
+                json.dump(self.hand_pose_msgs, f)
+            self.log.info(f"Created hand pose file: {hand_pose_file}")
+        else:
+            self.log.info(f"Skipping hand pose file creation")
+
+        if self.extract_head_pose_data:
+            head_pose_file = self.data_folder + "head_pose_data.txt"
+            with open(head_pose_file, mode="w", encoding='utf-8') as f:
+                json.dump(self.head_pose_msgs, f)
+            self.log.info(f"Created head pose file: {head_pose_file}")
+        else:
+            self.log.info(f"Skipping head pose file creation")
+
+        if self.extract_spatial_map_data:
+            spatial_map_file = self.data_folder + "spatial_map_data.txt"
+            with open(spatial_map_file, mode="w", encoding='utf-8') as f:
+                json.dump(self.spatial_map_msgs, f)
+            self.log.info(f"Created spatial map file: {spatial_map_file}")
+        else:
+            self.log.info(f"Skipping spatial map file creation")
+
+    def print_bag_info(self) -> None:
+        """
+        Print stats about the bag.
+        """
+        self.log.info(f"Bag info for {self.bag_path}:")
+        self.log.info(f"Total ROS messages: {self.num_total_msgs}")
+        self.log.info(f"Audio messages: {self.num_audio_msgs}")
+        self.log.info(f"Eye gaze messages: {self.num_eye_gaze_msgs}")
+        self.log.info(f"Hand pose messages: {self.num_hand_pose_msgs}")
+        self.log.info(f"Head pose messages: {self.num_head_pose_msgs}")
+        self.log.info(f"Image messages: {self.num_image_msgs}")
+        self.log.info(f"Spatial map messages: {self.num_spatial_map_msgs}")
+
+    def create_wav_file(self, filename: Optional[str] = None) -> None:
+        """
+        Creates a .wav file from the bag's audio data topic messages.
+        """
+        if not self.extract_audio:
```

Comment:
This seems to be the only post-collection function that has it's enable-protection inside the function. It's not wrong, just seems to not be self-consistent with the other bits of functionality here.